### DISCUSSION TITLE
Is there a way to load a particular extractor in middleware (I'm guessing yes)?

### DISCUSSION THREAD

**Discussion by thomasmost** [tokio-rs/axum#1870] [Q&A] [answered]
I know how to compose extensions in middleware, i.e. an 'admin' middleware can expect a 'session_user' extension... but is there a way to check a Query extractor in middleware?

Instead of loading a Query, I am currently doing this:

```rs
pub async fn current_org<B>(mut req: Request<B>, next: Next<B>) -> Result<Response, StatusCode> {
  if let Some(full_path) = req.uri().path_and_query() {
    let re = Regex::new(r"(.*)?org=([a-zA-Z0-9\-]*)").unwrap();
    let uri_str = full_path.to_string();
    if let Some(captures) = re.captures(&uri_str) {
      let org_id = captures.get(2).map_or("", |m| m.as_str());
    ...
```

...but it would obviously be easier to do something like this:

```rs

  let Some(query) = req.extensions().get::<Query<OrgId>>() else {
      return Err(
        StatusCode::BAD_REQUEST
      );
    };

```

...but this doesn't work (I've tried it). Which makes sense, since an 'extracted' Query is *not* a request extension, as I understand axum's model

**Comment by davidpdrsn** [comment, marked as answer]
https://docs.rs/axum/latest/axum/middleware/fn.from_fn.html#running-extractors
